In [5]:
import pandas as pd
import numpy as np

In [6]:
import pandas as pd

# Reading the CSV files
d1 = pd.read_csv("student-mat.csv", sep=";")
d2 = pd.read_csv("student-por.csv", sep=";")

# Merging the DataFrames on specified columns
merge_columns = ["school", "sex", "age", "address", "famsize", "Pstatus", "Medu", "Fedu", "Mjob", "Fjob", "reason", "nursery", "internet"]
d3 = pd.merge(d1, d2, on=merge_columns)

# Print the number of rows in the merged DataFrame
print(f"Number of students: {len(d3)}")
d3.head()

Number of students: 382


school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home     other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home     other  ...   
3     GP   F   15       U     GT3       T     4     2   health  services  ...   
4     GP   F   16       U     GT3       T     3     3    other     other  ...   

  famrel_y freetime_y  goout_y  Dalc_y  Walc_y health_y absences_y G1_y G2_y  \
0        4          3        4       1       1        3          4    0   11   
1        5          3        3       1       1        3          2    9   11   
2        4          3        2       2       3        3          6   12   13   
3        3          2        2       1       1        5          0   14   14   
4        4          3        2       1       2        5          0   11   13   

  G3_y  
0   11  
1   11  
2   12  
3   14  
4   13  

[5 rows x 53 columns]

In [7]:
print(d3.columns)

Index(['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu',
       'Mjob', 'Fjob', 'reason', 'guardian_x', 'traveltime_x', 'studytime_x',
       'failures_x', 'schoolsup_x', 'famsup_x', 'paid_x', 'activities_x',
       'nursery', 'higher_x', 'internet', 'romantic_x', 'famrel_x',
       'freetime_x', 'goout_x', 'Dalc_x', 'Walc_x', 'health_x', 'absences_x',
       'G1_x', 'G2_x', 'G3_x', 'guardian_y', 'traveltime_y', 'studytime_y',
       'failures_y', 'schoolsup_y', 'famsup_y', 'paid_y', 'activities_y',
       'higher_y', 'romantic_y', 'famrel_y', 'freetime_y', 'goout_y', 'Dalc_y',
       'Walc_y', 'health_y', 'absences_y', 'G1_y', 'G2_y', 'G3_y'],
      dtype='object')


Hypotheses for One-Tail Test

Null Hypothesis (H0): There is no difference in final grades between students with and without internet access, or students without internet access score higher than or equal to those with internet.

Alternative Hypothesis (H1): Students with internet access score higher on average than those without.

In [9]:
# testing math grades 
from scipy.stats import ttest_ind

# Splitting the dataset based on internet access for the math course
grades_with_internet_x = d3[d3['internet'] == 'yes']['G3_x']
grades_without_internet_x = d3[d3['internet'] == 'no']['G3_x']

# Performing the t-test (two-tailed by default)
t_stat_x, p_value_x = ttest_ind(grades_with_internet_x, grades_without_internet_x)

# For a one-tail test (assuming we're testing if internet access results in higher grades), 
# you would halve the p-value if the t-statistic is positive.
one_tail_p_value_x = p_value_x / 2 if t_stat_x > 0 else p_value_x

print(f'Math Two-Tail: t-statistic={t_stat_x:.3f}, p-value={p_value_x:.3f}')
print(f'Math One-Tail: t-statistic={t_stat_x:.3f}, one-tail p-value={one_tail_p_value_x:.3f}')


Math Two-Tail: t-statistic=1.722, p-value=0.086
Math One-Tail: t-statistic=1.722, one-tail p-value=0.043


Interpretation
Two-Tail Test: The p-value of 0.086 suggests that, under a two-tailed test with a typical alpha level of 0.05, you would fail to reject the null hypothesis. This means there is not enough evidence to suggest a statistically significant difference in math final grades between students with and without internet access when considering both possible directions of difference.

One-Tail Test: However, when considering the one-tail test, the halved p-value of 0.043 falls below the 0.05 threshold. Given that the t-statistic is positive, this indicates that you can reject the null hypothesis in favor of the alternative hypothesis that students with internet access at home have higher math final grades than those without, but only in the context of a one-tailed test.

What This Means
For the math subject (G3_x), the one-tailed test suggests a statistically significant effect of having internet access on improving final grades, under the specific hypothesis that internet access would lead to higher grades. This result could be of interest to educators and policymakers looking to understand the impact of digital resources on student learning outcomes.

However, it's crucial to approach one-tailed tests with caution and ensure they're justified based on prior evidence or a strong theoretical rationale, as they inherently involve testing for an effect in a predetermined direction. The result here underscores the potential value of internet access as a resource for academic support, particularly in math, suggesting that efforts to increase internet accessibility might positively impact student performance.

Given the findings, educational stakeholders might consider policies or interventions aimed at increasing home internet access for students as part of broader strategies to support math learning and overall academic achievement.

In [10]:
# Splitting the dataset based on internet access for the Portuguese course
grades_with_internet_y = d3[d3['internet'] == 'yes']['G3_y']
grades_without_internet_y = d3[d3['internet'] == 'no']['G3_y']

# Performing the t-test (two-tailed by default)
t_stat_y, p_value_y = ttest_ind(grades_with_internet_y, grades_without_internet_y)

# For a one-tail test if internet access results in higher grades),
one_tail_p_value_y = p_value_y / 2 if t_stat_y > 0 else p_value_y

print(f'Portuguese Two-Tail: t-statistic={t_stat_y:.3f}, p-value={p_value_y:.3f}')
print(f'Portuguese One-Tail: t-statistic={t_stat_y:.3f}, one-tail p-value={one_tail_p_value_y:.3f}')


Portuguese Two-Tail: t-statistic=1.743, p-value=0.082
Portuguese One-Tail: t-statistic=1.743, one-tail p-value=0.041


Interpretation
Two-Tail Test: The p-value of 0.082 for the two-tailed test suggests that, at the conventional alpha level of 0.05, you would not reject the null hypothesis for Portuguese. This means there is not enough evidence to suggest a statistically significant difference in final grades between students with and without internet access when considering potential differences in both directions.

One-Tail Test: The one-tail p-value of 0.041, however, is below the 0.05 threshold, and given the positive t-statistic, this indicates that you can reject the null hypothesis in favor of the alternative hypothesis for a one-tailed test. This suggests that students with internet access have significantly higher final grades in Portuguese compared to those without, under the specific hypothesis that internet access would lead to higher grades.

Conclusion
For the Portuguese subject (G3_y), similar to the math subject, the one-tailed test suggests that having internet access is associated with a statistically significant improvement in academic performance. This effect is consistent across both subjects, highlighting the potential value of internet access in supporting students' learning and academic achievements.

Given these findings, educational strategies that ensure equitable access to internet resources could be beneficial, especially in supporting students' academic success in key subjects like Portuguese. It underscores the importance of digital resources in modern education and could guide policy decisions regarding technology access in educational settings.

This analysis demonstrates the importance of considering the directionality of your hypothesis and choosing the appropriate statistical test to accurately interpret the relationship between variables of interest, such as internet access and academic performance.





